tortoise/ninja_tortoise.go

```diff
@@ -533,7 +534,7 @@ func (ni *ninjaTortoise) handleIncomingLayer(newlyr *types.Layer) { //i most rec
 			}
 
 			complete := true
-			for idx := windowStart; idx < j; idx++ {
+			for idx := types.LayerID(0); idx < j; idx++ {
```

Comment:
why start from layer 0? We shouldn't hold any info on layers outside the window, so why not start from the bottom of the window? Perhaps I'm missing the purpose of this iteration.

Reply:
true this was a temporary solution, we should only save the decision on blocks outside of window, we need to clean the maps and save a boolean regarding contextual validity

Reply:
of course this part has a lot to do with the layer roleback and maybe we should think  about it considering the changes we still have to make let me know how you want to proceed with this

Reply:
#803